### ISSUE TITLE
Distance + API- clarification

### ISSUE THREAD

**Issue by yoniker** [ZhaoJ9014/face.evoLVe.PyTorch#86]
Hey guys!

Thanks again for the great work.
So looking into the code here's what I will love to know:

Does taking the normalized output and then doing the L2 distance between the normalized outputs is almost always the right thing to do? (I'm asking because that's what's being done in verification.py@evaluate which calls verification.py@calculate_roc which does l2 distance, and the embeddings are always normalized as well)?

Also does the following code snippet represent the way that you intended the user code of face recognition to be?

```
def l2_norm(input, axis = 1):
    norm = torch.norm(input, 2, axis, True)
    output = torch.div(input, norm)
    return output

net=IR_50(input_size=[112,112]).cuda()
net.load_state_dict(torch.load('./backbone_ir50_asia.pth'))


def get_embeddings(net,image):
        import torchvision.transforms as transforms
        from torch.autograd import Variable
        import torch.nn
        PIC_SIZE=112
        transform=transforms.Compose([transforms.Resize((PIC_SIZE,PIC_SIZE)),transforms.RandomHorizontalFlip(),transforms.ToTensor(),transforms.Normalize((0.5,0.5,0.5),(0.5,0.5,0.5))])
        transformed_image=transform(image)
        the_image=Variable(transformed_image).cuda().unsqueeze(0)
        net.eval()
        outputs=net.forward(the_image)
        outputs=l2_norm(
        embeddings=outputs.detach().cpu().numpy()
        return embeddings

def calc_distance(embeddings1, embeddings2):
    diff = np.subtract(embeddings1, embeddings2)
    dist = np.sum(np.square(diff), 1)
    return dist
brad1=Image.open('aligned/Brad Pitt/Brad Pitt/1.jpg')
brad2=Image.open('aligned/Brad Pitt/Brad Pitt/2.jpg')
jennifer1= Image.open('aligned/Brad Pitt/Jennifer Aniston/1.jpg')
jennifer2= Image.open('aligned/Brad Pitt/Jennifer Aniston/1.jpg')
brad1_emb=get_embeddings(net,brad1)
brad2_emb=get_embeddings(net,brad2)
jennifer1_emb=get_embeddings(net,jennifer1)
jennifer2_emb=get_embeddings(net,jennifer2)
#And now just calc_distance between any two embeddings
```

**Comment by xsacha** [comment]
Yes, this is how the author intends the distance calculation.
There may be better ways of calculating it but this is a typical way.
